### ISSUE TITLE
Mode lost after saving the file to the server

### ISSUE THREAD

**Issue by pwwang** [mkloubert/vscode-remote-workspace#4]
The mode is `777` before I save it through VSCode:
```bash
-rwxrwxrwx 1 user group 1.9K May 25 14:20 proc.py
```
After I save it, it's changed to `666`:
```shell
-rw-rw-rw- 1 user group 1.9K May 25 14:23 proc.py
```
The executable permission is lost. Can we try to keep it?

**Comment by mkloubert** [comment]
@pwwang 

**Since version 0.22.0** there is a new parameter, called `mode`:

* **You can define an octal value**, which will be applied for all files and folder, like `mode=644`
* **or you can define a path to a JSON file**, which contains the mappings as object:

```json
{
    "644": "**/*.py",
    "777": [ "/*.txt" ]
}
```

The last matching [pattern](https://github.com/isaacs/minimatch) wins.

Save your mappings to a file, like `sftp_mappings.json` inside your home directory, e.g., and setup the parameter with the path to that file, like `mode=sftp_mappings.json`.

Relative paths will be mapped to the user's home directory.

**Comment by mkloubert** [comment]
@pwwang 

**I released a new version 0.23.0**, which does try to keep the permission value of an existing file by default now.

This can be controlled by new `keepMode` parameter.
